core/src/main/java/org/infinispan/commands/functional/Mutations.java

```diff
@@ -18,46 +20,72 @@ private Mutations() {}
 
    // No need to occupy externalizer ids when we have a limited set of options
    static <K, V, R> void writeTo(ObjectOutput output, Mutation<K, V, R> mutation) throws IOException {
+      BaseMutation bm = (BaseMutation) mutation;
+      DataConversion.writeTo(output, bm.keyDataConversion);
+      DataConversion.writeTo(output, bm.valueDataConversion);
       byte type = mutation.type();
       output.writeByte(type);
       switch (type) {
          case ReadWrite.TYPE:
             output.writeObject(((ReadWrite<K, V, ?>) mutation).f);
             break;
          case ReadWriteWithValue.TYPE:
-            output.writeObject(((ReadWriteWithValue<K, V, R>) mutation).value);
-            output.writeObject(((ReadWriteWithValue<K, V, R>) mutation).f);
+            ReadWriteWithValue<K, V, R> rwwv = (ReadWriteWithValue<K, V, R>) mutation;
```

Comment:
Is there any way to refactor the code so we just need to do an if ? instead of checking every type to write f everytime etc ...

Reply:
Do you want me to remove the switch? I  could add `writeTo` method to each mutation, instead of calling `.type()`, but I prefer to keep the writeTo/readFrom symmetric (and readFrom cannot be refactored that way).

Reply:
It's not a big deal for me, code seemed to me very similar in the switch so I suggested a change, but do whatever you think is better/homogenous for this case.